src/blueprints/tv2/tv2-on-timeline-generate-calculator.ts

```diff
@@ -0,0 +1,288 @@
+import { Tv2BlueprintConfiguration } from './value-objects/tv2-blueprint-configuration'
+import { Tv2ShowStyleBlueprintConfiguration } from './value-objects/tv2-show-style-blueprint-configuration'
+import { SisyfosChannelsTimelineObject, SisyfosType } from '../timeline-state-resolver-types/sisyfos-types'
+import { OnTimelineGenerateResult } from '../../model/value-objects/on-timeline-generate-result'
+
+const ACTIVE_GROUP_PREFIX: string = 'active_group_'
+const LOOK_AHEAD_GROUP_ID: string = 'look_ahead_group'
+const PREVIOUS_GROUP_PREFIX: string = 'previous_group_'
+
+export class Tv2OnTimelineGenerateCalculator implements BlueprintOnTimelineGenerate {
+  constructor(private readonly sisyfosPersistentLayerFinder: Tv2SisyfosPersistentLayerFinder) {}
+
+  public onTimelineGenerate(
+    configuration: Configuration,
+    previousRundownPersistentState: RundownPersistentState,
+    activePart: Part,
+    previousPart: Part | undefined,
+    timeline: Timeline
+  ): OnTimelineGenerateResult {
+    const blueprintConfiguration: Tv2BlueprintConfiguration = {
+      studio: configuration.studio.blueprintConfiguration as Tv2StudioBlueprintConfiguration,
+      showStyle: configuration.showStyle.blueprintConfiguration as Tv2ShowStyleBlueprintConfiguration
+    }
+
+    const rundownPersistentState: Tv2RundownPersistentState = this.convertToTv2RundownPersistentState(previousRundownPersistentState)
+    const newRundownPersistentState: Tv2RundownPersistentState = {
+      activeMediaPlayerSessions: [],
+      isNewSegment: previousPart?.segmentId !== activePart.segmentId,
+    }
+
+    this.assignSisyfosPersistMetaData(newRundownPersistentState, activePart, previousPart, timeline)
+
+    newRundownPersistentState.activeMediaPlayerSessions = this.assignMediaPlayerSessions(rundownPersistentState.activeMediaPlayerSessions, timeline, blueprintConfiguration)
+
+    return { rundownPersistentState: newRundownPersistentState, timeline }
+  }
+
+  private convertToTv2RundownPersistentState(persistentState: RundownPersistentState): Tv2RundownPersistentState {
+    let rundownPersistentState: Tv2RundownPersistentState = persistentState as Tv2RundownPersistentState
+    if (!rundownPersistentState) {
+      rundownPersistentState = {
+        activeMediaPlayerSessions: []
+      }
+    }
+    return rundownPersistentState
+  }
+
+  private assignSisyfosPersistMetaData(newRundownPersistentState: Tv2RundownPersistentState, activePart: Part, previousPart: Part | undefined, timeline: Timeline): void {
+    if (!newRundownPersistentState.isNewSegment || this.isAnySisyfosPieceInjectedIntoPart(activePart)) {
+      const sisyfosPersistedLevelsTimelineObject: TimelineObject =
+          this.createSisyfosPersistedLevelsTimelineObject(activePart, previousPart, newRundownPersistentState)
+      const activeTimelineObjectGroup: TimelineObjectGroup | undefined = timeline.timelineGroups.find(
+        (timelineObject) => timelineObject.id.includes('active_group_')
```

Comment:
Not any good reason.

I want us to be able access the different groups on the Timeline more easily than `timeline.timelineGroups.filter(...)` (something like `timeline.activeGroup` would be nice), but I haven't gotten around to it yet.

I'm planning on doing it before finishing the entirity of SOF-1539, but I'm postponing towards the end so I have an idea of how the `Timeline` is being used by the system to give me a better idea of how to implement it.

The use of magic string here instead of a const is most likely because I forgot to use a const...